### ISSUE TITLE
Adding https support

### ISSUE THREAD

**Issue by derxe** [FooSoft/anki-connect#90]
I'm trying to use AnkiConnect on https website and it wont let me connect since [https pages can only access other htttps resources](https://stackoverflow.com/questions/4032104/http-ajax-request-via-https-page). 

I'm just asking if adding https support to the script would be acceptable. In other words would you accept pull request with this feature.

I still not 100% if it is going to solve my problem, but it would be still nice to know if the feature is going to be accepted, so that I don't work on something that is not going to be used.

**Comment by Maecenas** [comment]
It is true that you cannot get a certificate for `localhost` from most of the certificate vendors like [Let's Encrypt](https://letsencrypt.org/). However, it is always the case that we want https on `localhost` when developing the service. To do this, follow the advice on starting an HTTPS server for [Python 2](https://gist.github.com/dergachev/7028596) and/or [Python 3](https://gist.github.com/elfgoh/8edccf02aaa357217721f227d555c8d4) since AnkiConnect is mostly in Python.

BTW, [Let's Encrypt](https://letsencrypt.org/) (aka [certbot](https://certbot.eff.org/)) is open-source, free and awesome for production. See also [`acme.sh`](https://github.com/Neilpang/acme.sh) for automating production.

**Comment by derxe** [comment]
Yes it is a website ([sentencesearch.neocities.org](https://sentencesearch.neocities.org/)), and since it uses https it cannot access ankiConnect with http and even if ankiConnect supported https it wouldn't be able to provide a valid certificate and thus still unable to connect. 

I never considered making an extension but it makes sense. The connection seems impossible without user's intervention and installing extension seems like an easy task for user to do. 

Thanks for suggestion :)

**Comment by FooSoft** [comment]
Cool site! 

Yeah I think your best shot is to create an extension; the user could then manage all of their deck settings and everything through there (much like what happens in Yomichan). The content script could interact with elements in your website and the integration pretty seamless.

I think that realistically that will be the easiest/most realistic thing to implement.
